scripts/bump_major_version.py

```diff
@@ -0,0 +1,335 @@
+    # as "documentation" of all the variant usages.
+    new_in_vXXX_string_00 = '<span class="version-tag">New in ' + \
+        version_string_name(version) + ':</span>'
+    new_in_vXXX_string_01 = '<span class="version-tag">New in ' + \
+        version_string_name(version) + '</span>:'
+    new_in_vXXX_string_02 = '<span class="version-tag">New in ' + \
+        version_string_name(version) + '</span>'
+    changed_in_vXXX_string_00 = '<span class="version-tag">Changed in ' + \
+        version_string_name(version) + '</span>'
+    changed_in_vXXX_string_01 = '<span class="version-tag">Changed in ' + \
+        version_string_name(version) + ':</span>'
+
+    # Read in the file
+    filedata = None
+    try:
+        with open(file, 'r') as f:
+            filedata = f.read()
+    except UnicodeDecodeError:
+        return
+
+    # Replace the target string
+    filedata = filedata.replace(new_in_vXXX_string_00, '')
+    filedata = filedata.replace(new_in_vXXX_string_01, '')
+    filedata = filedata.replace(new_in_vXXX_string_02, '')
+    filedata = filedata.replace(changed_in_vXXX_string_00, '')
+    filedata = filedata.replace(changed_in_vXXX_string_01, '')
+
+    # Write the file out again
+    with open(file, 'w') as f:
+        f.write(filedata)
+
+
+def update_image_versions(old_version, new_version, file):
+    # Int Int Path -> IO!
+
+    old_image_string = '''img src="{{ 'images/''' + \
+        version_string_name(old_version)
+    new_image_string = '''img src="{{ 'images/''' + \
+        version_string_name(new_version)
+
+    # Read in the file
+    filedata = None
+    try:
+        with open(file, 'r') as f:
+            filedata = f.read()
+    except UnicodeDecodeError:
+        return
+
+    # Replace the target string
+    filedata = filedata.replace(old_image_string, new_image_string)
+
+    # Write the file out again
+    with open(file, 'w') as f:
+        f.write(filedata)
+
+
+def grep_for_old_versionish_things(old_version, file):
+    # Int Path -> IO!
+    pat = re.compile(version_string_name(old_version))
+    lineno = 0
+    try:
+        with open(file) as f:
+            for line in f:
+                lineno += 1
+                matches = re.findall(pat, line)
+                if len(matches) != 0:
+                    print("{}:{}".format(file, lineno))
+                    print("{}".format(line))
+    except UnicodeDecodeError:
+        pass
+
+
+def update_base_yaml_config(to_version):
+    # Int -> IO!
+    file = os.path.join(docs_directory, '_config_base.yml')
+
+    # Read in the file
+    filedata = None
+    try:
+        with open(file, 'r') as f:
+            filedata = f.read()
+    except UnicodeDecodeError:
+        return
+
+    config = yaml.load(filedata, Loader=yaml.BaseLoader)
+
+    new_version_name = version_string_name(to_version)
+
+    newdata = '''
+    name: {}
+    version: {}
+    docker_image: cockroachdb/cockroach-unstable
+    build_time: 2020/05/12 11:00:26 (go1.13.4)
```

Comment:
Does this mean the script has to be updated on each use with similar hardcoded values?

Reply:
I don't think so.  At least the `docker_image` field is always `cockroachdb/cockroach-unstable` at a time when this script would be run (early in a new version's release cycle).

Re: the `build_time` field, it looks like it has been getting copied forward from one release to the next in the YAML config for a while.  I don't know what uses it, but apparently it has been Fine So Far (tm).

(... until it's not? I have not seen any breakage from it in the past - famous last words)